superset/views/core.py

```diff
@@ -1234,7 +1231,7 @@ def testconn(  # pylint: disable=too-many-return-statements,no-self-use
         uri = request.json.get("uri")
         try:
             if app.config["PREVENT_UNSAFE_DB_CONNECTIONS"]:
-                check_sqlalchemy_uri(uri)
+                check_sqlalchemy_uri(make_url(uri))
```

Comment:
I'm surprised we weren't getting mypy errors here before

Reply:
Me too.